misc/python/materialize/mzcompose.py

```diff
@@ -263,6 +264,9 @@ def get_workflow(
 
         built_steps = []
         for raw_step in workflow["steps"]:
+            # A step could be reused over several workflows, so operate on a copy
+            raw_step = raw_step.copy()
```

Comment:
This probably needs a `deepcopy` -- that or changing `raw_step.pop`  below should address the issue that you ran into? Do you remember how this failed before you made this change?

Reply:
On the first iteration the ```raw_step``` it is something like:

```
{'step': 'workflow', 'workflow': 'start-deps'}
```

and then it becomes:

```
{'workflow': 'start-deps'}
```

I considered dealing with the ```pop()``` but there is more code further down the line that expects that the ```step``` item has already been popped out. Also, in the longer term it would be safer to operate on a copy, e.g. if we implement additional variable substitution features.

As this is a flat data structure doing a deep copy did not seem to be needed.

Reply:
Thanks for the explanation!